src/AElf.OS.Core/Network/Application/ISyncStateService.cs

```diff
@@ -0,0 +1,182 @@
+
+        public SyncState GetSyncState()
+        {
+            switch (_syncStateProvider.SyncTarget)
+            {
+                case 0:
+                    return SyncState.UnInitialized;
+                case -1:
+                    return SyncState.Finished;
+                default:
+                    return SyncState.Syncing;
+            }
+        }
+        
+        private void SetSyncTarget(long value) => _syncStateProvider.SetSyncTarget(value);
+
+        /// <summary>
+        /// Based on current peers, will determine if a sync is needed or not. This method
+        /// should only be called once, to go from an uninitialized state to either syncing
+        /// or not syncing.
+        /// </summary>
+        /// <returns></returns>
+        public async Task StartSyncAsync()
+        {
+            if (GetSyncState() != SyncState.UnInitialized)
+            {
+                Logger.LogWarning("Trying to start the sync, but it has already been started/finished.");
+                return;
+            }
+
+            await TryFindSyncTargetAsync();
+        }
+
+        /// <summary>
+        /// Updates the current target for the initial sync. For now this method will
+        /// not have any effect if the sync is already finished or the target has not
+        /// been initialized.
+        /// </summary>
+        /// <returns></returns>
+        public async Task UpdateSyncStateAsync()
+        {
+            // This method should only be called when the sync target has already been found and the
+            // node is syncing.
+            
+            if (GetSyncState() != SyncState.Syncing)
+            {
+                Logger.LogWarning("Trying to update the sync, but it is either finished or not yet been initialized.");
+                return;
+            }
+                
+            var chain = await _blockchainService.GetChainAsync();
+            
+            // if the current LIB is higher than the recorded target, update
+            // the peers current LIB height. Note that this condition will 
+            // also be true when the node starts.
+            if (chain.LastIrreversibleBlockHeight >= _syncStateProvider.SyncTarget)
+            {
+                // Update handshake information of all our peers
+                var tasks = _peerPool.GetPeers().Select(async peer =>
+                {
+                    try
+                    {
+                        await peer.UpdateHandshakeAsync();
+                    }
+                    catch (NetworkException e)
+                    {
+                        Logger.LogError(e, "Error while updating the lib.");
+                    }
+                    
+                    Logger.LogDebug($"Peer {peer} last known LIB is {peer.LastKnowLibHeight}.");
+                    
+                }).ToList();
+                
+                await Task.WhenAll(tasks);
+                await TryFindSyncTargetAsync();
```

Comment:
when you call TryFindSyncTargetAsync multiple times in every method's end, 
it must be something wrong

Reply:
Hum, I see what you mean, but here is not a problem because the start method is only called once (node init) after that only update is called.